notion/block.py

```diff
@@ -629,6 +634,24 @@ def views(self):
             return None
         return [self._client.get_collection_view(view_id, collection=self.collection) for view_id in self.get("view_ids")]
 
+    def add_new_view(self, type="table"):
+        if not self.collection:
+            logging.warning("{} does not have 'collection_id' set; skipping.".format(self))
+            return None
+
+        view = self._client.get_collection_view(
+            self._client.create_record("collection_view", parent=self, type=type),
+            collection=self._collection
+        )
+        view.set("collection_id", self._collection.id)
+        view_ids = self.get("view_ids", [])
+        view_ids.append(view.id)
+        self.set("view_ids", view_ids)
+
+        # At this point, the view does not see to be completely initialized yet.
+        # Need to wait a second or two before e.g. setting a query.
+        return view
```

Comment:
waiting a second or two after this allows setting queries to work, so something async seems to be happening here